src/Internal/Cardano/TextEnvelope.purs

```diff
@@ -71,32 +80,43 @@ newtype TextEnvelope =
 
 derive instance Newtype TextEnvelope _
 
-data TextEnvelopeDecodeError
-  = JsonDecodeError JsonDecodeError
-  | CborParseError CborParseError
+decodeCborHexToBytes :: String -> Maybe ByteArray
+decodeCborHexToBytes cborHex = do
+  cborBa <- hexToByteArray cborHex
+  hush $ toByteArray $ wrap $ wrap cborBa
 
 decodeTextEnvelope
-  :: Aeson -> Either TextEnvelopeDecodeError TextEnvelope
+  :: Aeson -> Maybe TextEnvelope
 decodeTextEnvelope aeson = do
   { "type": type_, description, cborHex } <-
-    lmap JsonDecodeError $ decodeAeson aeson :: _ TextEnvelopeRaw
-  cborBa <- note (JsonDecodeError $ TypeMismatch "Hex") $ hexToByteArray cborHex
-  -- NOTE PlutusScriptV1 is doubly-encoded cbor, so the resulting `ByteArray`
-  -- is *still* cbor encoded.
-  -- https://github.com/Emurgo/cardano-serialization-lib/issues/268#issuecomment-1042986055
-  ba <- lmap CborParseError $ toByteArray $ wrap $ wrap cborBa
+    hush $ decodeAeson aeson :: _ TextEnvelopeRaw
+  ba <- decodeCborHexToBytes cborHex
   pure $ wrap { type_, description, bytes: ba }
 
-printTextEnvelopeDecodeError :: TextEnvelopeDecodeError -> String
-printTextEnvelopeDecodeError = case _ of
-  JsonDecodeError err -> printJsonDecodeError err
-  CborParseError err -> show err
-
 textEnvelopeBytes
-  :: String -> TextEnvelopeType -> Either TextEnvelopeDecodeError ByteArray
+  :: String -> TextEnvelopeType -> Maybe ByteArray
 textEnvelopeBytes json ty = do
-  aeson <- lmap JsonDecodeError $ parseJsonStringToAeson json
+  aeson <- hush $ parseJsonStringToAeson json
   TextEnvelope te <- decodeTextEnvelope aeson
-  unless (te.type_ == ty) $ throwError $ JsonDecodeError $ TypeMismatch $
-    show ty
-  pure te.bytes
+  if (te.type_ == ty) then pure te.bytes else Nothing
+
+_plutusScriptVxFromEnvelope
+  :: TextEnvelopeType
+  -> (ByteArray -> PlutusScript)
+  -> String
+  -> Maybe PlutusScript
+_plutusScriptVxFromEnvelope version bytesToScript str = do
```

Comment:
Within our conventions/code style we use underscores only for FFI functions and lenses.